pkg/component/component.go

```diff
@@ -116,6 +174,12 @@ func (c *Component) Type() string {
 	return ""
 }
 
+// Model components model
+type Model struct {
```

Comment:
Please add comments to this struct describing how it is used and the effects of each field. How does `Model.Signed.Data` related to `Model.Components`? Where does the signature in `Signed` come from, and which components use it? At what stage(s) is the signature verified -- do consumers of the `Components` field have any obligation to check anything?

Reply:
Added.

Reply:
Thank you, this helps a lot. Some followup questions: What form of the policy is signed / verified? For example:
- Some policies use variable substitution. If a policy field has `${some_variable}` in it, will that substitution break the signature?
- Policies can include conditions that are required for certain inputs or config subtrees, and if those conditions evaluate to false it can remove units or pieces of their config. Even if the condition passes, its raw text representation within the policy is altered. Does this break the signature?

Related: this description looks like the signature covers the full component list, is that right or does it split up componentwise? There are various things that might block a specific input (aside from the conditions mentioned above there may be platform restrictions / preventions, `capabilities.yml` blocks on certain input types, etc). If an unrelated input is removed from the model, would that break the signature from Endpoint's perspective?

Reply:
Only small portion of the data from the policy is signed, in this care only data to support tamper protection use cases, for example:
```
{
    "id": "de8cf520-11fa-11ee-87c3-87540e933c0c",
    "agent": {
        "features": {
            "tamper_protection": {
                "enabled": true
            }
        },
        "protection": {
            "enabled": false,
            "uninstall_token_hash": "hF8DWoc2hdb9jm0NAtNwQFVpb6FUCtkIquS/IDtO8jE=",
            "signing_key": "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEptaH31UPWD4b9mo+W8uzKnjN+w3aIWMmoxuR3qQsXs8oRfKQ+8AhdDHHMIWCeNz++HAxu5J3srPAq3vobND6wQ=="
        }
    },
    "inputs": [
        {
            "id": "5be5a402-b3cb-456a-92e7-5ca5ac9204af",
            "name": "EP",
            "revision": 1,
            "type": "endpoint"
        }
    ]
}
```
The reason the whole policy can't be signed because some policy fields are already updated by the fleet server.

Replacement of the values in the policy will not not break the signature, since the data (```signed.data``` field) is serialized and signed before the policy saved in Elasticsearch by kibana and is never modified between kibana and Endpoint for example. 

Over the time we can carefully sign more data from the policy, but we would have to continue support the the legacy behavior for some time, thus will not be able to protect the whole policy from tampering.